### DISCUSSION TITLE
API - prefixes - available-ips

### DISCUSSION THREAD

**Discussion by resparis** [netbox-community/netbox#8675] [Q&A] [answered]
Hi All, 

Im trying to get available ips on an specific prefix and it is returning an empty list. On the GUI, I can see 2 availables:

API GET:
http://<ip>/api/ipam/prefixes/13/available-ips/

![image](https://user-images.githubusercontent.com/49069898/154481520-0471b156-30e6-4b55-936f-aa063a656cdf.png)

![image](https://user-images.githubusercontent.com/49069898/154481685-614420fc-22d6-4da7-afa4-ffb11e7d1b7a.png)

![image](https://user-images.githubusercontent.com/49069898/154481800-bdd65b93-4466-4d8b-8478-5c0b1f3e96ac.png)

Any ideas on why is this happen?

**Comment by markkuleinio** [comment, marked as answer]
Your /31 subnet is already fully allocated (two addresses), so there is nothing /available-ips/ could return.

**Reply by resparis** [reply]
Thanks! I did add all prefixes AND subnets in advance ... Great!
